subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1144,16 +1144,16 @@ private void preventIllegalMutation(MutationType type) {
         if (resolvedState == ARTIFACTS_RESOLVED) {
             // The public result for the configuration has been calculated.
             // It is an error to change anything that would change the dependencies or artifacts
-            throw new InvalidUserDataException(String.format("Cannot change %s of %s after it has been resolved.", type, getDisplayName()));
+            throw new InvalidUserDataException(String.format("Cannot change %s of dependency %s after it has been resolved.", type, getDisplayName()));
```

Comment:
Not too thrilled about doing it that way, but any change to `displayName` is a different adventure, most likely requiring to make the difference between buckets, `canBeConsumed` and `canBeResolved` configurations.

Reply:
I think it is ok. If you use the display name and have more context around it, saying only _configuration_ might be enough because it is clear from the context that we talk about a _dependency configuration_.

Alternatively, you could not use the display name at all here and construct the whole message from the name of the configuration. Which might make the code a bit 'nicer'. (But the current solution is ok imo.)